test/controllers/income/estimatedPay/update/IncomeUpdateCalculatorControllerSpec.scala

```diff
@@ -766,11 +1075,13 @@ class IncomeUpdateCalculatorControllerSpec
   "handleBonusPayments" must {
     "redirect the user to bonusOvertimeAmountPage page" when {
       "user selected yes" in {
-        val testController = createTestIncomeUpdateCalculatorController
-        val cacheMap = Map(UpdateIncome_BonusPaymentsKey -> YesValue)
-        when(journeyCacheService.cache(Matchers.eq(cacheMap))(any())).thenReturn(Future.successful(Map("" -> "")))
-        val result = testController.handleBonusPayments()(
-          RequestBuilder.buildFakeRequestWithAuth("POST").withFormUrlEncodedBody(YesNoChoice -> YesValue))
+
+        val result = UnifiedHarness
+          .setupHandleBonusPayments()
+          .handleBonusPayments(
+            RequestBuilder.buildFakeRequestWithAuth("POST")
```

Comment:
This is a very common pattern in this file
```
RequestBuilder.buildFakeRequestWithAuth("POST").withFormUrlEncodedBody(YesNoChoice -> YesValue)
```
Consider encapsulating this code in the request builder. Something like
```
object RequestBuilder {
  def postRequest(formArgs: (String, String)*): Request = {
    buildFakeRequestWithAuth("POST") .withFormUrlEncodedBody(formArgs: _*)
  }
}
```
and usage could be simplified to
```
RequestBuilder.postRequest(YesNoChoice -> YesValue)
```

Reply:
Fair - I wasn't entirely sure whether or not it would be an obfuscation.